### PR TITLE
Adds a migration to reset the Total Issuance value

### PR DESCRIPTION
**Pull request by unconst**
The general process is the following:

1. Fix the incorrect places where a burn is occuring in the code.
2. Add a migration which will re set the total issuance by summing up the balances, stake values and lock values.
3. Adds tests here for all three types of account balances.

### CODE REVIEW

File: pallets/subtensor/src/utils.rs [hunk 1]
```diff
@@ -294,6 +294,9 @@ impl<T: Config> Pallet<T> {
     pub fn burn_tokens(amount: u64) {
         TotalIssuance::<T>::put(TotalIssuance::<T>::get().saturating_sub(amount));
     }
+    pub fn mint_tokens(amount: u64 ){
```
Comment (shibshib) [inline pallets/subtensor/src/utils.rs]: This sort of naming might freak people out. Maybe let's just rename it to `update_totalissuance` or something

**Review by sam0x17 — APPROVED** [review]
LGTM 🎉

File: pallets/subtensor/src/block_step.rs [hunk 2]
```diff
@@ -148,9 +148,9 @@ impl<T: Config> Pallet<T> {
                     &Self::get_subnet_owner(netuid),
                     Self::u64_to_balance(cut.to_num::<u64>()).unwrap(),
                 );
-                TotalIssuance::<T>::put(
-                    TotalIssuance::<T>::get().saturating_add(cut.to_num::<u64>()),
-                );
+
+                // We are minting tokens here for the owner.
+                Self::mint_tokens( cut.to_num::<u64>() );
```
Comment (shibshib) [inline pallets/subtensor/src/block_step.rs]: Same comment as below, 'minting' should be reworded to something less defi-sounding.

File: pallets/subtensor/src/registration.rs [hunk 3]
```diff
@@ -401,9 +401,10 @@ impl<T: Config> Pallet<T> {
 
         // --- 5. Add Balance via faucet.
         let balance_to_add: u64 = 100_000_000_000;
+        Self::mint_tokens( 100_000_000_000 ); // Accounting here for the newly created tokens.
```
Comment (shibshib) [inline pallets/subtensor/src/registration.rs]: Why is the total issuance being updated here?

Comment (unconst) [reply pallets/subtensor/src/registration.rs]: because this is the faucet.

File: pallets/subtensor/src/block_step.rs (hunk 2 above)
Comment (unconst) [reply pallets/subtensor/src/block_step.rs]: ok. legal.

File: pallets/subtensor/src/utils.rs (hunk 1 above)
Comment (unconst) [reply pallets/subtensor/src/utils.rs]: ok big brother

Comment (sam0x17) [reply pallets/subtensor/src/utils.rs]: `update_total_issuance` imo

File: pallets/subtensor/src/migration.rs [hunk 4]
```diff
@@ -22,6 +22,46 @@ pub mod deprecated_loaded_emission_format {
...
+        weight = weight.saturating_add(T::DbWeight::get().reads_writes(Stake::<T>::iter().count() as u64, 0));
+
+        // Calculate the sum of all stake values
+        let locked_sum: u64 = SubnetLocked::<T>::iter()
+            .fold(0, |accumulator, (_, locked_value)| accumulator.saturating_add(locked_value));
+        weight = weight.saturating_add(T::DbWeight::get().reads_writes(SubnetLocked::<T>::iter().count() as u64, 0));
+
+        // Retrieve the total balance sum
+        let total_balance = T::Currency::total_issuance();
+        let total_balance_sum: u64 = total_balance.try_into().unwrap_or_else(|_| panic!("Conversion must be within range"));
```
Comment (sam0x17) [inline pallets/subtensor/src/migration.rs:48]: honestly this panic is really scary, could we just keep with the type of `T::Currency::total_issuance()` instead or does it not support the ops we need? (another reason to use my currencies crate in the future hahaha)

**Review by sam0x17 — CHANGES_REQUESTED** [review]

**Review by sam0x17 — APPROVED** [review]

**Review by shibshib — APPROVED** [review]

**Pull request merged**